lib/FrontendTool/FrontendTool.cpp

```diff
@@ -184,20 +185,39 @@ static bool emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
 }
 
 namespace {
+struct SwiftModuleTraceInfo {
+  std::string ModulePath;
+  bool IsImportedDirectly;
+  bool IsCompiledWithLibraryEvolution;
+};
+
 struct LoadedModuleTraceFormat {
   std::string Name;
   std::string Arch;
-  std::vector<std::string> SwiftModules;
+  std::vector<SwiftModuleTraceInfo> SwiftModules;
 };
 }
 
 namespace swift {
 namespace json {
+template <> struct ObjectTraits<SwiftModuleTraceInfo> {
+  static void mapping(Output &out, SwiftModuleTraceInfo &contents) {
+    out.mapRequired("path", contents.ModulePath);
+    out.mapRequired("is-imported-directly", contents.IsImportedDirectly);
+    out.mapRequired("is-compiled-with-library-evolution",
+                    contents.IsCompiledWithLibraryEvolution);
+  }
+};
+
 template <> struct ObjectTraits<LoadedModuleTraceFormat> {
   static void mapping(Output &out, LoadedModuleTraceFormat &contents) {
     out.mapRequired("name", contents.Name);
```

Comment:
While we're here, let's add a version field. Everything should have version fields.

Reply:
What kind of versioning scheme do you have in mind? The Swift compiler version? Have a separate "trace format version" using the Basic/Version class? Something else?

Reply:
Trace format version, which can be "1" or "2" at this point. Again, just to make it easier to potentially change things in the future.